autoprotocol/protocol.py

```diff
@@ -6745,10 +6748,109 @@ def location_helper(source, destination, volume, method, density):
+
+            Returns
+            -------
+            list(Informatics)
+                list of Informatics per destination well
+
+            Raises
+            ------
+            TypeError
+                Informatics provided is not valid or not supported
+            ValueError
+                Informatics wells must match wells in Instruction
+            ValueError
+                Multiple instances of Informatics on a Well
+            ValueError
+                Parsed list of Informatics length should match the length of destination
+            """
+            if multi_src:
+                dest_count = 1
+            else:
+                dest_count = len(dest)
+
+            if len(informatics) == 1:
+                if isinstance(informatics[0], AttachCompounds):
+                    compounds = informatics[0].compounds
+                    wells = WellGroup(informatics[0].wells)
+                    if len(wells) == dest_count and set(wells) == set(destination):
+                        info_compd = compounds * dest_count
+                        informatics_list = []
+                        for well, compd in zip(destination, info_compd):
+                            if isinstance(informatics[0], AttachCompounds):
+                                if not isinstance(compd, list):
+                                    compd = [compd]
+                                if not isinstance(compd, list):
+                                    compd = [compd]
+                                informatics_list.append(AttachCompounds(well, compd))
+                    else:
+                        raise ValueError(
+                            f"Informatics wells: {wells} do not match wells used in Instruction."
+                        )
+                else:
+                    raise TypeError(
+                        f"Informatics:{informatics} is not available in this protocol."
+                    )
+                # If transfer has multiple sources and one destination, Informatics should be added
+                # to the last transfer.
+                if multi_src:
+                    informatics_list = [None] * (len(dest) - 1) + informatics
+            else:
+                wells_compounds_dict = {}
+                for info in informatics:
+                    if isinstance(info, AttachCompounds):
+                        wells_count = len(WellGroup(info.wells))
+                        compounds = info.compounds * wells_count
+                        for w, c in zip(WellGroup(info.wells).wells, compounds):
+                            if w not in wells_compounds_dict.keys():
+                                wells_compounds_dict[w] = c
+                            else:
+                                raise ValueError(
+                                    f"There are multiple instances of Informatics for well: {w}. The "
+                                    f"intent is ambiguous."
+                                )
```

Comment:
hmm... this may be overly strict (its okay on the first-pass, could leave a TODO if you don't want to tackle this now).

For this case
```
p.transfer(
a.well(0), [a.well(1), a.well(1)], "10:microliter", 
informatics=[AttachCompounds(a.well(1), [CompoundA]), AttachCompounds(a.well(1), [CompoundB])
)
```
It seems reasonable to expect this to be valid, i.e. I would expect there to be a Compound attached at the completion of each liquid handle instruction.

Reply:
I see. That makes sense. I'll revise on this PR.